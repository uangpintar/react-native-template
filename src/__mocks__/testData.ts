export const products = {
  products: [
    {
      id: 1,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 2,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 3,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 4,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
  ],
};

export const users = {
  users: [
    {
      id: 1,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
    {
      id: 2,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
    {
      id: 3,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
  ],
};
