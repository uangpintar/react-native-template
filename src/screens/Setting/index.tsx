import { useNavigation } from '@react-navigation/native';
import { Button } from 'native-base';
import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';

export default function Setting() {
  const navigation = useNavigation();
  return (
    <SafeAreaView style={{ flex: 1, padding: 20, justifyContent: 'center' }}>
      <Button onPress={() => navigation.navigate('RemoteConfigScreen')}>Remote Config Demo</Button>
    </SafeAreaView>
  );
}
