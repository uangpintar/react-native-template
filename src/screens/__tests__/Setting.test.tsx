import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import Setting from '../Setting';
import { wrapper } from '../../__mocks__/wrapper';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));

describe('Setting screen', () => {
  it('Render Setting screen', async () => {
    const setting = render(<Setting />, { wrapper });

    const btnSubmit = setting.getByTestId('btn-demo-rcd');
    expect(btnSubmit).toBeTruthy();
    fireEvent.press(btnSubmit);

    await waitFor(() => expect(mockedNavigate).toHaveBeenCalledTimes(1));
    expect(mockedNavigate).toHaveBeenCalledWith('RemoteConfigScreen');
    expect(setting.toJSON()).toMatchSnapshot();
  });
});
