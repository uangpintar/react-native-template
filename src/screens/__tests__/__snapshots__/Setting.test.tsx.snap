// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting screen Render Setting screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
        "left",
        "right",
        "top",
      ]
    }
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0891b2",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 10,
        }
      }
      testID="btn-demo-rcd"
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
        test={true}
      >
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Remote Config Demo
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaProvider>
`;
